### PR TITLE
Add e2e test case fo check --pre

### PR DESCRIPTION
**Pull request by chanwit**
This PR implements an e2e test case for the `flux check --pre` happy path.

To this end, this PR adds a templating mechanism for golden files.

### CODE REVIEW

**Comment by chanwit** [comment]
@stefanprodan @hiddeco could you please take a look?

File: cmd/flux/check_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/fluxcd/flux2/internal/utils"
+)
+
+func TestCheckPre(t *testing.T) {
+	out, err := utils.ExecKubectlCommand(context.TODO(), utils.ModeCapture, rootArgs.kubeconfig, rootArgs.kubecontext, "version", "--short")
```
Comment (stefanprodan) [inline cmd/flux/check_test.go:14]: Please use `kubectl version --output json` then use the json objects to extract the versions.

File: cmd/flux/check_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	kubectlVersion := strings.TrimPrefix(versions["clientVersion"].GitVersion, "v")
+	k8sVersion := strings.TrimPrefix(versions["serverVersion"].GitVersion, "v")
+
+	cmd := cmdTestCase{
+		args:            "check --pre",
+		wantError:       false,
+		testClusterMode: ExistingClusterMode,
+		templateValues: map[string]string{
+			"kubectlVersion": kubectlVersion,
```
Comment (stefanprodan) [inline cmd/flux/check_test.go:34]: Let's rename these to `clientVersion` and  `serverVersion` same in the goldfile please.

File: cmd/flux/main_test.go [hunk 3]
```diff
@@ -203,14 +206,32 @@ func (cmd *cmdTestCase) runTestCmd(t *testing.T) {
 
 	var expected string
 	if cmd.goldenValue != "" {
-		expected = cmd.goldenValue
+		if cmd.templateValues != nil {
+			tmpl := template.Must(template.New("golden").Parse(cmd.goldenValue))
```
Comment (stefanprodan) [inline cmd/flux/main_test.go:210]: Can you please extract this to a func so we don't have to duplicate the template code.

Comment (chanwit) [reply cmd/flux/main_test.go:210]: got it

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @chanwit 🏅

**Pull request merged**